src/main/java/com/minecolonies/core/compatibility/jei/JobBasedRecipeCategory.java

```diff
@@ -58,14 +59,13 @@
  * Base class for a JEI recipe category that displays a Minecolonies citizen based on a job.
  //* @param <T> The recipe type.
  */
-public abstract class JobBasedRecipeCategory<T> implements IRecipeCategory<T>
+public abstract class JobBasedRecipeCategory<T> extends AbstractRecipeCategory<T>
 {
+    private static final Map<EquipmentTypeEntry, List<ItemStack>> TOOL_CACHE = new HashMap<>();
```

Comment:
Ideally we should cache this more globally, but that should probably be separate from this PR anyway.